internal/dockerignore/ignore_test.go

```diff
@@ -0,0 +1,76 @@
+package dockerignore
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/windmilleng/tilt/internal/ignore"
+	"github.com/windmilleng/tilt/internal/testutils"
+)
+
+func TestIsIgnored(t *testing.T) {
+	tf := newTestFixture(t, "node_modules")
+	tf.AssertResult(tf.JoinPath("node_modules", "foo"), true, false)
+	tf.AssertResult(tf.JoinPath("foo", "bar"), false, false)
+}
+
+func TestComment(t *testing.T) {
+	tf := newTestFixture(t, "# generated code")
+	tf.AssertResult(tf.JoinPath("node_modules", "foo"), false, false)
+	tf.AssertResult(tf.JoinPath("foo", "bar"), false, false)
+}
+
+func TestGlob(t *testing.T) {
+	tf := newTestFixture(t, "*/temp*")
+	tf.AssertResult(tf.JoinPath("somedir", "temporary.txt"), true, false)
+	tf.AssertResult(tf.JoinPath("somedir", "temp"), true, false)
+}
+
+func TestOneCharacterExtension(t *testing.T) {
+	tf := newTestFixture(t, "temp?")
+	tf.AssertResult(tf.JoinPath("tempa"), true, false)
+	tf.AssertResult(tf.JoinPath("tempeh"), false, false)
+	tf.AssertResult(tf.JoinPath("temp"), false, false)
+}
+
+type testFixture struct {
+	repoRoot *testutils.TempDirFixture
+	t        *testing.T
+	tester   ignore.Tester
+}
+
+func newTestFixture(t *testing.T, dockerignores ...string) *testFixture {
+	tf := testFixture{}
+	tempDir := testutils.NewTempDirFixture(t)
+	tf.repoRoot = tempDir
+	ignoreText := strings.Builder{}
+	for _, rule := range dockerignores {
+		ignoreText.WriteString(rule + "\n")
+	}
+	tempDir.WriteFile(".dockerignore", ignoreText.String())
+
+	tester, err := NewDockerfileIgnoreTester(tempDir.Path())
+	if err != nil {
+		t.Fatal(err)
+	}
+	tf.tester = tester
+
+	tf.t = t
+	return &tf
+}
+
+func (tf *testFixture) JoinPath(path ...string) string {
+	return tf.repoRoot.JoinPath(path...)
+}
+
+func (tf *testFixture) AssertResult(path string, expectedIsIgnored bool, expectError bool) {
+	isIgnored, err := tf.tester.IsIgnored(path, false)
+	if expectError {
```

Comment:
ARE you testing any cases where you expect an error if you can think of some, might be worth it?

Reply:
No, I think I should just remove it. Looking at the code coverage of this test all the errors are IO related, even the ones inside of the library.